internal/bft/controller.go

```diff
@@ -595,112 +595,101 @@ func (c *Controller) sync() (viewNum uint64, seq uint64, decisions uint64) {
 
 	if len(syncResponse.RequestDel) != 0 {
 		c.RequestPool.Prune(func(bytes []byte) error {
-			return errors.New("need all delete")
+			return errors.New("need all delete") // TODO why are all requests deleted here?
```

Comment:
The entire concept of RequestDel is flawed, it shouldn't be used. 

Fabric itself should not even populate this field, so in practice the library skips it. 

I think we need to remove this, in general.

Reply:
I see that it has been indeed [removed](https://github.com/SmartBFT-Go/consensus/pull/559) from the official repo.

Reply:
Then I'll leave this for another PR
Maybe I'll open an issue about removing this